### ISSUE TITLE
Notification 'red' badge count is flawed in E2EE rooms

### ISSUE THREAD

**Issue by ara4n** [vector-im/element-web#17223]
You can only correctly calculate badge counts for mentions & keyword push rules if you decrypt every message in an E2EE room on the client.  Currently if there is a gappy sync (due to the app being offline overnight, or the client failing to keep up with the server) then messages deliberately get dropped.  Therefore the 'red' badge count for missed notifications is wrong (and the NotifPanel is wrong) whenever the app awakes in the morning - which is when you want the badge counts to be the most correct.

Instead we should spider the E2EE rooms in the background in order to calculate the right notification count.  Luckily we have the code to do this already for Seshat - but we need to actually hook it up on both Web & Desktop so that it triggers badge count calculations as it spiders, so we at last get the correct badge counts.

cc @manuroe and @bmarty for iOS & Android, as the same problem exists there (but is less pronounced given gappy syncs are somewhat less common given people tend not to turn off or disconnect their phones as often as they do their laptops).

cc @gsouquet as this overlaps somewhat with his decrypt-on-demand work in https://github.com/matrix-org/matrix-js-sdk/pull/1684

**Comment by ara4n** [comment]
(is it really minor that your notifications plain lie about whether you've been notified or not?)

**Comment by babolivier** [comment]
I also disagree with this issue being categorised as minor. Since I close Element when I'm not working (which I assume a fair amount of people using Element for work would do), this means that I'll need to go through each of my encrypted rooms manually to make sure I haven't missed a notification at least once a week (after the weekend) and each time I come back after some time off. IMHO this doesn't paint a great picture of Element in terms of reliability if using it for work means it's likely to drop notifications on the floor at least once a week.

**Comment by MadLittleMods** [comment]
(from https://github.com/vector-im/element-web/issues/9069) Also related to https://github.com/matrix-org/matrix-doc/pull/1796

**Comment by richvdh** [comment]
> Instead we should spider the E2EE rooms in the background in order to calculate the right notification count. 

For links: this also came up in the context of discussions around MSC3952 (https://github.com/matrix-org/matrix-spec-proposals/pull/3952#discussion_r1114284205).

**Comment by richvdh** [comment]
Closely related: https://github.com/vector-im/element-web/issues/6874.
